shelley/chain-and-ledger/formal-spec/update.tex

```diff
@@ -1,26 +1,140 @@
-\footnote{as opposed to waiting until the end of an epoch},
-they are handled separately.
-The adoption of new application versions is what triggers software updates.
-Therefore $\mathsf{UPDATE}$ is the combination of the
-$\mathsf{PPUP}$ transition for the protocol parameters and the
-the $\mathsf{AVUP}$ transition for the application versions.
-Note that $\PPUpdate$, $\AVUpdate$, and $\Update$ were defined in
-Figure~\ref{fig:defs:utxo-shelley}, and that $\fun{pvCanFollow}$
-was defined in \cite{byron_ledger_spec}.
-
+The governance process includes a mechanism for core nodes to propose and vote on
+updates. In this chapter we
+outline rules for genesis keys \textit{proposing} both protocol parameter
+and application version updates, as well as voting on whether a particular
+software update is an acceptable future option (again, only genesis keys vote on this).
+For rules regarding the \textit{adoption} of protocol
+parameter updates, see \ref{sec:pparam-update}. For rules regarding
+adoption of new software versions see \ref{sec:software-updates}. \newline
+
+\noindent \underline{Distinction between protocol parameter and application updates.}
+\begin{itemize}
+\item \textbf{Protocol parameters} (excluding $\ProtVer$, explained below): constants
+currently used in ledger calculations according to the rules described in this document
+\begin{itemize}
+\item[$\circ$] Stored on the leger
+\item[$\circ$] Changed only at epoch boundary
+\item[$\circ$] Changed according to the genesis key votes, which are also stored on the ledger
+\item[$\circ$] All nodes automatically adopt new values (this mechanism is an explicit
+ part of the ledger rules, and is included in the epoch boundary transition)
+\item[$\circ$] Has no effect on ledger transition rules
+\end{itemize}
+\item \textbf{Protocol version} ($\ProtVer$): a special value in PP's which
+corresponds to a specific version of the \textit{ledger rules}
+\begin{itemize}
+\item[$\circ$] I.e. if $\var{pv}$ changes, this document should be updated with
+the new rules
+\item[$\circ$] E.g. the change may be that the new rules now allow \textit{all} nodes
+to vote on update proposals
+\item[$\circ$] Adoption mechanism is the same as PP's (which this value is included it),
+so the new rules come into effect at an epoch boundary
+\item[$\circ$] Because a new set of transition rules must now be followed, nodes must have
+software installed that can implement these rules at the epoch boundary
+when the PP adoption occurs
+\item[$\circ$] Switching to using these new rules is mandatory in the sense that
+if the nodes do not have the applications implementing them, this
+may prevent a user from reading and
+writing to the blockchain, e.g changing the
+transaction data structure would cause such a problem
+\end{itemize}
+\item \textbf{Applications} The version of the software the nodes run,
+as well as the related system tags and metadata
+\begin{itemize}
+\item[$\circ$] These updates cannot be forced
```

Comment:
we cannot force the users to actually upgrade their software, but the data on the chain for the application versions is in a sense forced.